atlp-rwanda/atlp-devpulse-bn
# Ft tickets

### PR Description

This PR allows users to create tickets in case they face issues and get replies from the admin.

### Description of tasks that were expected to be completed

As an applicant I want to be able to open a ticket/inquiry about my application. This includes submitting a form with a title and a body explaining my inquiry. The tickets are sent to the admin (they can not be assigned to nor seen by another applicant).

Possible statuses:
- Open
- Admin-reply (when the admin has replied to it)
- applicant-reply (when the applicant has replied to the admin and is awaiting another response)
- Resolved (when the both parties have agreed that the issue has been resolved)

If the applicant is not happy with the resolved, they can reply again which changes the ticket status back to applicant-reply

Note: Only the ticket author and admin can view tickets.

### How has this been tested?
- Clone repo
- Checkout branch `ft-tickets`
- Make sure that you set the environment variables referring to `.env.example`
- Run `npm install`, then `npm run dev` to check if the app is running successfully
- Test the following queries and mutations at [localhost](http://localhost:5000/)

```
{
  query GetUserTickets 
  query GetAllTickets   // should be logged in as an admin
  query GetTicketById
  query FilterTicketByDetails
  mutation CreateTicket
  mutation ResolveTicket  // admin
  mutation updateTicket
}

```

### Track PR (issue number & link)
Issue: [#215](https://github.com/atlp-rwanda/atlp-devpulse-fn/issues/215)

### Screenshots (If appropriate especially on the frontend)
N/A
